pkg/gce-cloud-provider/compute/gce.go

```diff
@@ -83,8 +106,49 @@ func CreateCloudProvider(vendorVersion string) (*CloudProvider, error) {
 
 }
 
-func createBetaCloudService(vendorVersion string) (*beta.Service, error) {
-	client, err := newDefaultOauthClient()
+func generateTokenSource(configFile *ConfigFile) (oauth2.TokenSource, error) {
+	// Note: if tokenURL is nil, set tokenSource to DefaultTokenSource.
+	// This allows running gceCloud remotely.
+	if configFile != nil && configFile.Global.TokenURL != "" && configFile.Global.TokenURL != "nil" {
+		tokenSource := NewAltTokenSource(configFile.Global.TokenURL, configFile.Global.TokenBody)
+		glog.V(4).Infof("Using AltTokenSource %#v", tokenSource)
+		return tokenSource, nil
+	}
+
+	tokenSource, err := google.DefaultTokenSource(
+		context.Background(),
+		compute.CloudPlatformScope,
+		compute.ComputeScope)
+	if gac, ok := os.LookupEnv("GOOGLE_APPLICATION_CREDENTIALS"); ok {
```

Comment:
I know you didn't write this code but I got confused when reading it again, could you put a comment saying that `DefaultTokenSource` tries to use `GOOGLE_APPLICATION_CREDENTIALS` and thats why we spit out a warning if its not set.

Reply:
Yeah good call, I had to dig in a little deeper to answer this question too